### PR TITLE
feat: add `assert/is-same-booleanarray`

### PR DESCRIPTION
**Pull request by Jaysukh-409**
Resolves: Subtask of #2304 

## Description

> What is the purpose of this pull request?

This pull request:

-   This PR will add `@stdlib/assert/is-same-booleanarray` assert package.

## Related Issues

> Does this pull request have any related issues?

This pull request:

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Jaysukh-409** [comment]
This PR's CI will pass once we merge the #2419

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+</section>
+
+<!-- /.usage -->
+
+<section class="notes">
+
+## Notes
+
+-   In contrast to the strict equality operator `===`, the function distinguishes between `+0` and `-0` and treats `NaNs` as the [same value][@stdlib/assert/is-same-value].
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/README.md]:
```suggestion
```

This note is not applicable for boolean arrays.

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+<section class="related">
+
+* * *
+
+## See Also
+
+-   <span class="package-name">[`@stdlib/assert/is-booleanarray`][@stdlib/assert/is-booleanarray]</span><span class="delimiter">: </span><span class="description">test if a value is a BooleanArray.</span>
+-   <span class="package-name">[`@stdlib/assert/is-same-uint8array`][@stdlib/assert/is-same-uint8array]</span><span class="delimiter">: </span><span class="description">test if two arguments are both Uint8Arrays and have the same values.</span>
+-   <span class="package-name">[`@stdlib/assert/is-same-value`][@stdlib/assert/is-same-value]</span><span class="delimiter">: </span><span class="description">test if two arguments are the same value.</span>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/README.md]:
```suggestion
```

This section, as mentioned in the comment, is handled by tooling. It should not be manually populated for new packages.

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+[@stdlib/assert/is-same-value]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-same-value
+
+<!-- <related-links> -->
+
+[@stdlib/assert/is-booleanarray]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-booleanarray
+
+[@stdlib/assert/is-same-uint8array]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-same-uint8array
+
+<!-- </related-links> -->
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/benchmark/benchmark.length.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+	x = [];
+	y = [];
+	for ( i = 0; i < len; i++ ) {
+		v = Boolean( i%2 );
+		x.push( v );
+		y.push( v );
+	}
+	x = new BooleanArray( x );
+	y = new BooleanArray( y );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/benchmark/benchmark.length.js]:
```suggestion
	for ( i = 0; i < len; i++ ) {
		x.push( Boolean( i%2 ) );
	}
	x = new BooleanArray( x );
	y = new BooleanArray( x );
```

You only need one source array.

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/benchmark/benchmark.length.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+* Creates a benchmark function.
+*
+* @private
+* @param {PositiveInteger} len - array length
+* @returns {Function} benchmark function
+*/
+function createBenchmark( len ) {
+	var x;
+	var y;
+	var v;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/benchmark/benchmark.length.js]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/docs/repl.txt [hunk 6]
```diff
@@ -0,0 +1,35 @@
+
+{{alias}}( v1, v2 )
+    Tests if two arguments are both BooleanArrays and have the same values.
+
+    The function differs from the `===` operator in that the function treats
+    `-0` and `+0` as distinct and `NaNs` as the same.
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/docs/repl.txt]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/docs/types/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+// TypeScript Version: 4.1
+
+/**
+* Tests if two arguments are both BooleanArrays and have the same values.
+*
+* ## Notes
+*
+* -   The function differs from the `===` operator in that the function treats `-0` and `+0` as distinct and `NaNs` as the same.
+*
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/docs/types/index.d.ts]:
```suggestion
```

File: lib/node_modules/@stdlib/assert/is-same-booleanarray/test/test.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+		void 0,
+		null,
+		[],
+		{},
+		function noop() {},
+		typedarray( 10, 'float32' ),
+		typedarray( 10, 'int32' ),
+		typedarray( 10, 'float32' ),
+		typedarray( [ 2.0, 4.0, 6.0, 8.0 ], 'complex64' ),
+		typedarray( [ 0, 0, 0, 0 ], 'uint8' )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/assert/is-same-booleanarray/test/test.js]:
```suggestion
		typedarray( [ 0, 0, 0, 0 ], 'bool' )
	];
	y = [
		'abc',
		'boop',
		-5,
		-3.14,
		3.14,
		-0.0,
		0.0,
		false,
		true,
		void 0,
		null,
		[],
		{},
		function noop() {},
		typedarray( 10, 'float32' ),
		typedarray( 10, 'int32' ),
		typedarray( 10, 'float32' ),
		typedarray( [ 2.0, 4.0, 6.0, 8.0 ], 'complex64' ),
		typedarray( [ 0, 0, 0, 1 ], 'bool' )
```

Slightly more interesting test case given that we already have L72.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Jaysukh-409!

**Pull request merged**